meteor/server/api/ingest/rundownInput.ts

```diff
@@ -370,36 +371,54 @@ Meteor.startup(() => {
 	}
 })
 
+interface MediaObjectUpdatedIds {
+	rundownId: RundownId
+	segmentId: SegmentId
+}
+
 function onMediaObjectChanged(newDocument: MediaObject, oldDocument?: MediaObject) {
 	if (
 		!oldDocument ||
 		(newDocument.mediainfo?.format?.duration &&
 			oldDocument.mediainfo?.format?.duration !== newDocument.mediainfo.format.duration)
 	) {
-		const segmentsToUpdate = new Map<SegmentId, RundownId>()
 		const rundownIdsInStudio = Rundowns.find({ studioId: newDocument.studioId }, { fields: { _id: 1 } })
 			.fetch()
 			.map((rundown) => rundown._id)
 		Parts.find({
 			rundownId: { $in: rundownIdsInStudio },
 			'hackListenToMediaObjectUpdates.mediaId': newDocument.mediaId,
-		}).forEach((part) => {
-			segmentsToUpdate.set(part.segmentId, part.rundownId)
-		})
-		segmentsToUpdate.forEach((rundownId, segmentId) => {
-			lazyIgnore(
-				`updateSegmentFromMediaObject_${segmentId}`,
-				async () => updateSegmentFromCache(newDocument.studioId, rundownId, segmentId),
-				200
-			)
 		})
+			.map<MediaObjectUpdatedIds>((part) => {
+				return {
+					rundownId: part.rundownId,
+					segmentId: part.segmentId,
+				}
+			})
+			.filter(doesSegmentExistsInCache)
+			.forEach((mediaObjectUpdatedIds: MediaObjectUpdatedIds) => {
+				lazyIgnore(
+					`updateSegmentFromMediaObject_${mediaObjectUpdatedIds.segmentId}`,
+					async () => updateSegmentFromCache(newDocument.studioId, mediaObjectUpdatedIds),
```

Comment:
Even though `lazyIgnore` passes this function to `Meteor.setTimeout`, which is supposed to catch, I think it would be considered good practice to catch the exception ourselves (see: https://github.com/tv2/tv-automation-server-core/blob/eb6e053d7d88ec30a74e2dc3f96e5927466da4d9/meteor/server/api/ingest/packageInfo.ts#L41-L43), because we still shall assume something in there might throw. Filtering by `doesSegmentExistsInCache` at this stage does not even guarantee that `IngestDataCache` will still contain the document when this job eventually gets to execute.
I presume the root cause of the issue described in SOF-1144 (the unhandledRejection causing a crash) was already fixed elsewhere after release 42 - is that correct?

Reply:
I'm not aware if the root cause has been fixed elsewhere.

The root cause seem to be that `IngestDataCache` has had Segments deleted from it, where the "deleted" Segment still exists in the `Segment` collection (and the Segments Parts in the `Parts` collection).

I have been unsuccesfully in trying to recreate that scenario, which could indicate that the root cause has been fixed (or that it's a very tricky edgecase to hit).

I can wrap it in a try-catch if you prefer, but I don't think it will be necessary.

Reply:
Have you tried when the segment has gotten into the state of `orphaned: 'deleted'`? I expect that in this scenario the `IngestDataCache` for the segment is gone.  

Getting into this state should be:
* a part in the segment is playing
* inews says to delete segment
* sofie ingest deletes segment
* `IngestDataCache` changes are written back to the database
* during the commit stage, it sofie 'restores' the deleted segment and marks it `orphaned: 'deleted'`, as it is playing so cannot be deleted

Reply:
Updated

Reply:
I was able to replicate the error thanks to the steps from @Julusian 

@ianshade I have added the try-catch as you suggested